### PR TITLE
ref: use Sprintf as helper to accept type any for the key of WithValue

### PR DESCRIPTION
**Pull request by mdanialr**
## 📑 Description

Relate to https://github.com/goravel/framework/pull/628

@coderabbitai summary

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

**Review by devhaozi — DISMISSED** [review]

**Review by hwbrzzl — COMMENTED** [review]
Could you add some test cases with different keys?

File: context_test.go [hunk 1]
```diff
@@ -14,3 +14,16 @@ func TestContext(t *testing.T) {
 	assert.Equal(t, ctx.Value("Hello").(string), "world")
 	assert.Equal(t, ctx.Value("Hi").(string), "Goravel")
 }
+
+func TestContextWithCustomKeyType(t *testing.T) {
+	type customKeyType struct{}
+	var customKey customKeyType
+	var customKeyTwo customKeyType
```
Comment (hwbrzzl) [inline context_test.go:21]: Please add additional other type keys: int, string, etc.

Comment (hwbrzzl) [reply context_test.go:21]: We may merge these two test cases: `TestContext`, `TestContextWithCustomKeyType`.

**Review by hwbrzzl — COMMENTED** [review]
Great 👍  please update the PR of goravel/framework first, then I'll approve it.

File: context_test.go (hunk 1 above)
Comment (mdanialr) [reply context_test.go]: done in https://github.com/goravel/gin/pull/87/commits/3b44987393c32d31ef3c571661b054770f343cdd

**Comment by hwbrzzl** [comment]
This PR needs to be merged into the v1.2.x branch, and the change of goravel/framework has been merged, please run `go get github.com/goravel/framework@6543aeadf1bc39901cb02df96020a1f7aabbaa03`, goravel/fiber is the same.

**Comment by mdanialr** [comment]
all right, let me checkout from `v1.2.x` branch then and do PR again

**Pull request closed without merging**